fdbserver/ConfigBroadcaster.actor.cpp

```diff
@@ -152,54 +110,19 @@ class ConfigBroadcasterImpl {
 					te.detail("Op", "Clear");
 				}
 
-				reply.changes.push_back_deep(reply.changes.arena(), versionedMutation);
+				req.changes.push_back_deep(req.changes.arena(), versionedMutation);
 			}
 		}
-		req.reply.send(reply);
-		++successfulChangeRequest;
-	}
 
-	ACTOR static Future<Void> serve(ConfigBroadcaster* self,
-	                                ConfigBroadcasterImpl* impl,
-	                                ConfigBroadcastFollowerInterface cbfi) {
-		impl->actors.add(impl->consumer->consume(*self));
-		loop {
-			choose {
-				when(ConfigBroadcastFollowerGetSnapshotRequest req = waitNext(cbfi.getSnapshot.getFuture())) {
-					++impl->snapshotRequest;
-					ConfigBroadcastFollowerGetSnapshotReply reply;
-					for (const auto& [key, value] : impl->snapshot) {
-						if (matchesConfigClass(req.configClassSet, key.configClass)) {
-							reply.snapshot[key] = value;
-						}
-					}
-					reply.version = impl->mostRecentVersion;
-					TraceEvent(SevDebug, "ConfigBroadcasterGotSnapshotRequest", impl->id)
-					    .detail("Size", reply.snapshot.size())
-					    .detail("Version", reply.version);
-					req.reply.send(reply);
-				}
-				when(ConfigBroadcastFollowerGetChangesRequest req = waitNext(cbfi.getChanges.getFuture())) {
-					if (req.lastSeenVersion < impl->lastCompactedVersion) {
-						req.reply.sendError(version_already_compacted());
-						++impl->failedChangeRequest;
-						continue;
-					}
-					if (req.lastSeenVersion < impl->mostRecentVersion) {
-						impl->sendChangesReply(req, impl->mutationHistory);
-					} else {
-						TEST(req.lastSeenVersion > impl->mostRecentVersion); // Worker is ahead of ConfigBroadcaster
-						TraceEvent(SevDebug, "ConfigBroadcasterRegisteringChangeRequest", impl->id)
-						    .detail("Peer", req.reply.getEndpoint().getPrimaryAddress())
-						    .detail("MostRecentVersion", impl->mostRecentVersion)
-						    .detail("ReqLastSeenVersion", req.lastSeenVersion)
-						    .detail("ConfigClass", req.configClassSet);
-						impl->pending.addRequest(req);
-					}
-				}
-				when(wait(impl->actors.getResult())) { ASSERT(false); }
-			}
+		if (req.changes.size() == 0) {
+			return Void();
 		}
+
+		client.lastSeenVersion = mostRecentVersion;
+		req.mostRecentVersion = mostRecentVersion;
+		// TODO: Retry in event of failure
+		++successfulChangeRequest;
+		return success(client.broadcastInterface.getChanges.getReply(req));
```

Comment:
@sfc-gh-tclinkenbeard are retries actually necessary here? If the worker fails, it will re-register with the broadcaster when it starts back up. Otherwise, `getReply` says `Reliable at least once delivery: Eventually delivers request at least once and returns one of the replies if communication is possible.`

Reply:
It's possible that a single message could get lost, even if the worker doesn't die, so I think using `RequestStream::getReply` is still best here.